# .yaml config file
#
# The PID gains and controller settings must be saved in a yaml file that gets loaded
# to the param server via the roslaunch file (robocol_rover_control.launch).
robocol_rover:
  # Publish all joint states -----------------------------------
  # Creates the /joint_states topic necessary in ROS
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  
  # Wheel Velocity Controllers ---------------------------------------
  front_wheel_L_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_front_left_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}
  middle_wheel_L_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_middle_left_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}
  back_wheel_L_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_rear_left_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}
  front_wheel_R_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_front_right_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}
  middle_wheel_R_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_middle_right_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}
  back_wheel_R_joint_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_rear_right_joint
    pid: {p: 1.0, i: 0.0, d: 0.0}

  # Wheel Trees Position Controllers ---------------------------------------
  #suspension_arm_F_L_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_F_L_joint
    #pid: {p: 2200.0, i: 10.0, d: 10.0}
  #suspension_arm_B_L_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_B_L_joint
    #pid: {p: 4200.0, i: 10.0, d: 10.0}
  #suspension_arm_B2_L_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_B2_L_joint
    #pid: {p: 2200.0, i: 10.0, d: 10.0}

  #suspension_arm_F_R_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_F_R_joint
    #pid: {p: 2200.0, i: 10.0, d: 10.0}
  #suspension_arm_B_R_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_B_R_joint
    #pid: {p: 4200.0, i: 10.0, d: 10.0}
  #suspension_arm_B2_R_joint_position_controller:
    #type: effort_controllers/JointPositionController
    #joint: suspension_arm_B2_R_joint
    #pid: {p: 2200.0, i: 10.0, d: 10.0}




